tools/scripts/publish.plugin.ts

```diff
@@ -0,0 +1,47 @@
+import { CreateNodes, CreateNodesContext, readJsonFile } from '@nx/devkit';
+import { dirname, join } from 'node:path';
+import { ProjectConfiguration } from 'nx/src/config/workspace-json-project-json';
+
+export const createNodes: CreateNodes = [
+  '**/project.json',
+  (projectConfigurationFile: string, opts, context: CreateNodesContext) => {
+    const root = dirname(projectConfigurationFile);
+    const projectConfiguration: ProjectConfiguration = readJsonFile(
+      projectConfigurationFile,
+    );
+
+    const isPublishable = Boolean(projectConfiguration?.targets?.publish);
+    const { name: projectName } = projectConfiguration;
```

Comment:
The publishable projects should contain a `package.json`, right?
Wouldn't `**/package.json` be a better glob?
It would remove the need to keep an empty `publish` target.

Reply:
Also buildable targets have a `package.json`. I updated the PR description.

Reply:
Afterthought:

I will maybe use the target name `publishable` to identify publishable libs